clang/include/clang/Driver/Options.td

```diff
@@ -2675,10 +2675,16 @@ defm sycl_id_queries_fit_in_int: BoolFOption<"sycl-id-queries-fit-in-int",
   LangOpts<"SYCLValueFitInMaxInt">, DefaultTrue,
   PosFlag<SetTrue, [], "Assume">, NegFlag<SetFalse, [], "Do not assume">,
   BothFlags<[CC1Option, CoreOption], " that SYCL ID queries fit within MAX_INT.">>;
+def fsycl_device_obj_EQ : Joined<["-"], "fsycl-device-obj=">,
+  Flags<[CoreOption]>, Values<"spirv,llvmir">, HelpText<"Specify format of "
+  "device code stored in the resulting object. Valid values are: spirv, llvmir "
+  "(default)">;
 def fsycl_use_bitcode : Flag<["-"], "fsycl-use-bitcode">,
-  Flags<[CC1Option, CoreOption]>, HelpText<"Use LLVM bitcode instead of SPIR-V in fat objects">;
+  Alias<fsycl_device_obj_EQ>, AliasArgs<["llvmir"]>, Flags<[CoreOption]>,
+  HelpText<"Use LLVM bitcode instead of SPIR-V in fat objects">;
```

Comment:
Can emission of deprecation messages be inserted into `Driver.cpp` at this stage already?

Myself, I don't have strong preferences as to whether this should be done in this PR or in a separate one - merely avoiding the loss of this requirement should be sufficient.

Reply:
Deprecation needs to go through the deprecation/removal process before we can start emitting the diagnostic.  I will create an issue so we don't lose track.